### PR TITLE
docs: fix auth example

### LINKED ISSUE #4523: 'The expression evaluated to a falsy value' in GitHub authentication example
- Playwright Version: 1.6.2
- Operating System: Windows 10
- Node.js version: 14.15.1

I just took this [authentication example](https://github.com/microsoft/playwright/blob/master/docs/examples/authentication.js) and run it and it didn't work. So I played with it but couldn't make it work (TypeScript):
```
import { chromium, webkit } from "playwright";
import * as assert from "assert";

const verifyIsLoggedIn = async (page) => {
  await page.click('summary[aria-label="View profile and more"]');
  let details = await page.waitForSelector("details[open]");
  await page.screenshot({ path: `verify.png` });
  // await page.waitForSelector("details-menu");           // ====> this one never works!
  const yourProfile = await details.$(`text="Your profile"`);
  assert(yourProfile);                                     // ====> FAILS
  await page.screenshot({ path: `verify1.png` });
};

/**
 * In this script, we will login on GitHub.com through Chromium,
 * and reuse the login cookies state inside WebKit. This recipe can be
 * used to speed up tests by logging in once and reusing login state.
 *
 * Steps summary
 * 1. Login on GitHub.com in Chromium
 * 2. Export cookies from Chromium browser context
 * 3. Set cookies in WebKit browser context and verify login
 */

const account = { login: "", password: "" };

(async () => {
  // Create a Chromium browser context
  const crBrowser = await chromium.launch();
  const crContext = await crBrowser.newContext();
  const crPage = await crContext.newPage();
  crPage.setDefaultTimeout(6000);

  // Navigate and auto-wait on the page to load after navigation
  await crPage.goto("https://github.com/login");

  // Fill login form elements
  await crPage.fill('input[name="login"]', account.login);
  await crPage.fill('input[name="password"]', account.password);

  // Submit form and auto-wait for the navigation to complete
  await crPage.click('input[type="submit"]');
  await verifyIsLoggedIn(crPage);

  // Get cookies from Chromium browser context
  const cookies = await crContext.cookies();
  await crBrowser.close();

  // Create WebKit browser context and load cookies
  const wkBrowser = await webkit.launch();
  const wkContext = await wkBrowser.newContext();
  await wkContext.addCookies(cookies);

  // Navigate to GitHub.com and verify that we are logged in
  const wkPage = await wkContext.newPage();
  await wkPage.goto("https://github.com");
  await verifyIsLoggedIn(wkPage);
  await wkBrowser.close();
})();
```
But the `verify1.png` screenshot does show the menu item that we're looking for.

### PR DESCRIPTION
**Pull request by dgozman**
We should wait for selector instead of asserting it.

Fixes #4523.

### CODE REVIEW

File: docs/examples/authentication.js [hunk 1]
```diff
@@ -50,5 +50,5 @@ const account = { login: '', password: '' };
 
 const verifyIsLoggedIn = async (page) => {
   await page.click('summary[aria-label="View profile and more"]')
-  assert(await page.$(`text="Your profile"`));
+  assert(await page.waitForSelector(`text="Your profile"`));
```
Comment (JoelEinbinder) [inline docs/examples/authentication.js:53]: It doesn't really make sense to assert here right? waitForSelector is truthy or throws.
```suggestion
 await page.waitForSelector(`text="Your profile"`);
```

Comment (dgozman) [reply docs/examples/authentication.js:53]: That's correct, but it is logical to assert in the `verifyIsLoggedIn` function, so why not?

**Comment by alvipeo** [comment]
This is not fix. See my new message in the original issue. And please re-open it.

**Comment by alvipeo** [comment]
See this - [https://github.com/microsoft/playwright/issues/4530](https://github.com/microsoft/playwright/issues/4530).

**Pull request merged**